### ISSUE TITLE
FALSE POSITIVE: NP_ALWAYS_NULL ignores a false field on a fresh object

### ISSUE THREAD

**Issue by chisa22** [spotbugs/spotbugs#4271]
Hi, I found a false positive in SpotBugs 4.10.3 when the fresh Box does not escape and its field is assigned false.

**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs NP_ALWAYS_NULL`

**Minimal reproducer**

```java
final class SpotBugsFieldState {
  static final class Box { boolean reachable; }
  void test() {
    Box box = new Box();
    box.reachable = false;
    if (box.reachable) {
      Object object = null;
      object.toString();
    }
  }
}
```

**Reproduction command**

```bash
spotbugs -version
javac -d classes spotbugs-np-always-null-fp-field-state.java
spotbugs -textui -effort:max -low -visitors FindNullDeref classes
```

**Current behavior**

SpotBugs reports `NP_ALWAYS_NULL` at line 8.

**Expected behavior**

SpotBugs should not report `NP_ALWAYS_NULL` because the fresh Box does not escape and its field is assigned false.
